sycl/source/detail/event_impl.hpp

```diff
@@ -342,6 +364,8 @@ class event_impl {
   std::weak_ptr<ext::oneapi::experimental::detail::graph_impl> MGraph;
   /// Indicates that the event results from a command graph submission.
   bool MEventFromSubmittedExecCommandBuffer = false;
+  /// Store the executable command graph associated with this event, if any.
+  ext::oneapi::experimental::detail::exec_graph_impl *MExecGraph = nullptr;
```

Comment:
We're using a raw pointer here, which makes me think what happens if a user calls `event.ext_oneapi_get_profiling_info(node)` after the `command_graph<executable>` object which was submitted for `event` was destroyed?

Reply:
The implementation has been updated to use a weak_ptr instead of the raw pointer.